File: src/services/tokenServices.js
require("dotenv").config();
const moment = require("moment");
const fetch = require("node-fetch");

var { prepareConnection, insertData } = require("../../database/conection");

const messageError = process.env.MESSAGE_ERROR;
const urlCurl = process.env.URL_CURL

function validateData() {
  return new Promise((resol, reject) => {
    prepareConnection(12, "automation")
      .then((resolve) => {
        //Se resuelve la conexión y trae los datos
        console.log(resolve);
        if (resolve[0].date_use == null && resolve[0].date_create != null) {
          let res = validateMain(resolve[0].date_create, resolve[0].value);
          if (res == false) {
            reject(messageError);
          } else {
            resol(res);
          }
        } else {
          let res = validateMain(resolve[0].date_use, resolve[0].value);
          if (res == false) {
            reject(messageError);
          } else {
            resol(res);
          }
        }
      })
      .catch((reject) => {
        //Error en la conexión
        console.log(reject);
      });
  });
}

function callPython() {
  console.log("ENTRO POR LLAMADO AL PYTHON");
  fetch(urlCurl)
    .then((response) => response.json())
    .then((data) => {
      console.log(data);
    })
    .catch((error) => {
      console.log("Hubo un error:", error);
    });
}

function validateMain(dataDate, dataValue) {
  let resFunction = validateDateAndDate(dataDate);
  if (resFunction != false) {
    //Imprimir valor del token
    //console.log(`Token sirve -> ${resolve[0].VALUE}`);
    return dataValue;
  } else {
    //console.log("Token no sirve ");
    callPython();
    return false;
  }
}

function validateDateAndDate(fecha) {
  const fechaData = moment(fecha, "YYYY-MM-DD HH:mm:ss.SSSSSS");
  const fechaActual = moment();
  const diferenciaEnMinutos = fechaActual.diff(fechaData, "minutes");
  if (diferenciaEnMinutos > 10) {
    console.log(
      "La fechaActual es mayor a 10 minutos que la fecha del token Jira."
      //MANDAR A EJECUTAR LA FUNCIÓN QUE LLAME EL PY
    );
    return false;
  } else {
    console.log(
      "La fechaActual no es mayor a 10 minutos que la fecha del token Jira."
    );
    return true;
  }
}

function validateDataInsert(req) {
  //console.log(req)
  return new Promise((resol, rejec) => {
    insertData(req.fecha, req.token,13)
      .then((resolve) => {
        //Se resuelve la conexión y trae los datos
        //console.log(resolve);
        resol(resolve);
      })
      .catch((reject) => {
        //Error en la conexión
        //console.log(reject);
        rejec(reject);
      });
  });
}

module.exports = { validateData, validateDataInsert };

File: src/services/tokenServices.py
import os
import datetime
import requests
from database.conection import prepareConnection, insertData
from dotenv import load_dotenv

load_dotenv()

messageError = os.environ.get("MESSAGE_ERROR")
urlCurl = os.environ.get("URL_CURL")


def validateData():
    def validateMain(dataDate, dataValue):
        resFunction = validateDateAndDate(dataDate)
        if resFunction is not False:
            return dataValue
        else:
            callPython()
            return False

    def callPython():
        print("ENTRO POR LLAMADO AL PYTHON")
        response = requests.get(urlCurl)
        data = response.json()
        print(data)

    def validateDateAndDate(fecha):
        fechaData = datetime.datetime.strptime(fecha, "%Y-%m-%d %H:%M:%S.%f")
        fechaActual = datetime.datetime.now()
        diferenciaEnMinutos = (fechaActual - fechaData).total_seconds() / 60
        if diferenciaEnMinutos > 10:
            print("La fechaActual es mayor a 10 minutos que la fecha del token Jira.")
            return False
        else:
            print("La fechaActual no es mayor a 10 minutos que la fecha del token Jira.")
            return True

    try:
        resolve = prepareConnection(12, "automation")
        dataDate = resolve[0].date_create if resolve[0].date_use is None else resolve[0].date_use
        dataValue = resolve[0].value
        if dataDate is not None and dataValue is not None:
            res = validateMain(str(dataDate), str(dataValue))
            return res
        else:
            return messageError
    except Exception as e:
        print("Error en la conexión:", e)


def validateDataInsert(req):
    try:
        resolve = insertData(req['fecha'], req['token'], 13)
        return resolve
    except Exception as e:
        print("Error en la conexión:", e)


if __name__ == "__main__":
    result = validateData()
    print(result)
